src/lib/Liblog/pbs_log.c

```diff
@@ -858,7 +910,14 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 	if ((text == NULL) || (objname == NULL))
 		return;
 
-	now = time(NULL);	/* get time for message */
+	if (pbs_conf.pbs_log_highres_timestamp && (gettimeofday(&tp, NULL) != -1)) {
+		now = tp.tv_sec;
+		microsec = tp.tv_usec;
+	} else {
+		now = time((time_t *)0);	/* get time for message */
+		microsec = 0;
```

Comment:
You don't have to set it to 0 in this case. But, even if you keep it, I'm ok with it.